File: Regular/112. Path Sum/Solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} sum
 * @return {boolean}
 */
var hasPathSum = function(root, sum) {
    if (root != null) {
        let diff = sum - root.val;
        if (root.left == null && root.right == null) return !diff;
        return hasPathSum(root.left, diff) || hasPathSum(root.right, diff);
    }
    return false;
};

File: Regular/112. Path Sum/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean hasPathSum(TreeNode root, int sum) {
        if (root != null) {
            int diff = sum - root.val;
            if (root.left == null && root.right == null) return diff == 0;
            return hasPathSum(root.left, diff) || hasPathSum(root.right, diff);
        }
        return false;
    }
}
